modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -143,37 +145,45 @@ class ONNXCompiled {
     void run();
 };
 
+static void appendOpenVINOExecutionProvider(Ort::SessionOptions *session_options,
+                                            const cv::gapi::onnx::ep::OpenVINO &ov_ep) {
+     OrtOpenVINOProviderOptions options;
+     options.device_type = ov_ep.device_type.c_str();
+     options.cache_dir = ov_ep.cache_dir.c_str();
```

Comment:
is there any memory life time problems in this?

Reply:
I don't think so. `ov_ep` is taken from `ParamDesc` which lives longer than both `OrtOpenVINOProviderOptions` and `Ort::SessionOptions` and amost the same time as `Ort::Session`.

Reply:
Who owns the `ParamDesc`? Is it stored somewhere in GModel ?

Reply:
`ParamsDesc` is part of `ONNXCompiled` which is produced by `ONNXUnit` which is stored in `GModel`
https://github.com/opencv/opencv/blob/4.x/modules/gapi/src/backends/onnx/gonnxbackend.cpp#L412-L420
https://github.com/opencv/opencv/blob/4.x/modules/gapi/src/backends/onnx/gonnxbackend.cpp#L474-L481